File: src/main/Image.java
package main;

import java.io.File;
import java.util.List;

/**
 * Created by Shirobako on 10/1/2016.
 */
public interface Image {

    List<String> getTags();

    void setLocation(File location);

    File getLocation();

    String getRating();
}

File: Common/src/main/kotlin/com/darkmage530/birat/file/Image.kt
package com.darkmage530.birat.file

import arrow.core.raise.either

data class Image(val file: File,
                 val md5: Md5,
                 val dimensions: Dimensions) {

    companion object {
        operator fun invoke(file: java.io.File) = either {
            val fileWrapper = File(file).bind()
            val a = getMd5(fileWrapper).bind()
            val b = getDimensions(fileWrapper).bind()
            Image(fileWrapper, a, b)
        }.mapLeft {
            it
        }
    }
}

//    fun getTags(): List<String>
//    fun setLocation(location: com.darkmage530.birat.file.File)
//    fun getLocation(): com.darkmage530.birat.file.File
//    fun getRating(): String